File: B3WM.Shared/Entity/VolumeLevel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace B3WM.Shared.Entity
{
    public class VolumeLevel
    {
        public double Price { get; set; }
        public long Total { get; set; }
        public long BuyVolume { get; set; }
        public long SellVolume { get; set; }

        public long Delta => BuyVolume - SellVolume;
    }
}

File: B3WM.Flutter/lib/models/volume_level.dart
class VolumeLevel {
  final double price;
  final int total;
  final int buyVolume;
  final int sellVolume;

  int get delta => buyVolume - sellVolume;

  VolumeLevel({
    required this.price,
    required this.total,
    required this.buyVolume,
    required this.sellVolume,
  });

  factory VolumeLevel.fromJson(Map<String, dynamic> json) {
    return VolumeLevel(
      price: (json['price'] as num).toDouble(),
      total: (json['total'] as num).toInt(),
      buyVolume: (json['buyVolume'] as num).toInt(),
      sellVolume: (json['sellVolume'] as num).toInt(),
    );
  }

  Map<String, dynamic> toJson() => {
        'price': price,
        'total': total,
        'buyVolume': buyVolume,
        'sellVolume': sellVolume,
      };
}
